packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.tsx

```diff
@@ -55,13 +55,20 @@ export class MessageBar extends React.Component<IMessageBarProps, IMessageBarSta
   }
 
   private _getActionsDiv(): JSX.Element {
-    if (this.props.actions) {
-      return this.props.isMultiline ?
-        <div className='ms-MessageBar-actions'> { this.props.actions } </div> :
-        <div className='ms-MessageBar-actionsOneline'>
+    if (this.props.isMultiline) {
+      if (this.props.actions) {
+        return <div className='ms-MessageBar-actions'>
+          { this.props.actions }
+        </div>;
+      }
+    }
+    else {
+      if (this.props.actions || this.props.onDismiss) {
+        return <div className='ms-MessageBar-actionsOneline'>
```

Comment:
It would be cleaner to separate out the logic to render action buttons and Dismiss button. 
`_getActionsDiv()` : check if actions are available and whether its multiline or not
`_getDismissDiv()` : check if onDismiss is available and its single line